src/panels/calendar/dialog-calendar-event-detail.ts

```diff
@@ -122,16 +128,83 @@ class DialogCalendarEventDetail extends LitElement {
   }
 
   private _renderRruleAsText(value: string) {
-    // TODO: Make sure this handles translations
     try {
-      const readableText =
-        value === "" ? "" : RRule.fromString(`RRULE:${value}`).toText();
-      return html`<div id="text">${readableText}</div>`;
+      // From the backend we get "BY_WEEKDAY" section, which however cannot be read in via RRule.fromString().
```

Comment:
```suggestion
      // From the backend we get "BY_WEEKDAY" section, which however cannot be read in via RRule.fromString().
```

@allenporter Can you comment on this?Is the backend changing that, because even if the frontend is sending "BYDAY" I later get "BY_WEEKDAY" back from the backend.

Reply:
I will investigate. I know it is common to rename by day to by weekday in code to make it easier to understand as a developer, but my impression is that it always needs to go back to byday when used in the string ([like this in rrule.js](https://github.com/jakubroztocil/rrule/blob/6de89457da0e0fb0f0335a9e59acaac3e97abc6a/src/optionstostring.ts#L35)) so this is likely a bug in either `dateutil.rrul` or `ical.types.recur`. Either way, core can handle it.

Reply:
OK, yes, the local calendar return value is bogus:
```
{"start":{"dateTime":"2023-01-04T10:00:00-08:00"},"end":{"dateTime":"2023-01-03T11:00:00-08:00"},"summary":"Local recurrence example","description":null,"location":null,"uid":"77fd295a-7405-11ed-9574-066a07ffbaf5","recurrence_id":"20230104T100000","rrule":"FREQ=WEEKLY;COUNT=13;INTERVAL=1;BY_WEEKDAY=[WeekdayValue(weekday=<Weekday.MONDAY: 'MO'>, occurrence=None), WeekdayValue(weekday=<Weekday.WEDNESDAY: 'WE'>, occurrence=None)]"}]
```
It is returning the python object and not the recurrence rule string. I'll work on a fix.

Reply:
I was able to reproduce the issue and fix the bug in `ical` and bumping in https://github.com/home-assistant/core/pull/83254

Reply:
With that bump I am now able to remove the regex workaround as it now works perfectly without.